frontend/src/app/core/state/resource-collection.service.ts

```diff
@@ -96,6 +97,21 @@ export abstract class ResourceCollectionService<T> {
       ) as Observable<T>;
   }
 
+  /**
+   * Clear a collection key
+   * @param key Collection key to clear
+   */
+  clear(key:string):void {
+    this
+      .store
+      .update(
+        (store) => ({
```

Comment:
can we use here deconstruction, as we are not interested in altering anything in store beside `collections`.

```
({ collections }) => (
  {
    collections: omit(collections, key),
  }
)
```

Reply:
like 

```
function omit(obj: Record<string, unknown>, key: string) {
    // eslint-disable-next-line no-unused-vars
    const {[key]: foo, ...newO} = obj;
    return newO;
}
```